### ISSUE TITLE
Changing `XDG_DATA_HOME` and `XDG_CONFIG_HOME` before nvim command resutls error

### ISSUE THREAD

**Issue by skyraider256526** [MordechaiHadad/bob#93]
I am following this tutorial to learn how to configure neovim ([link](https://alpha2phi.medium.com/neovim-for-beginners-init-lua-45ff91f741cb)) with lua and there we are trying to create a isolated environment to not interfere with any existing configuration.
When I am trying this `XDG_DATA_HOME=$NVIM_BEGINNER/share XDG_CONFIG_HOME=$NVIM_BEGINNER nvim` it gives me following error:
`ERROR Error: The used file required for bob could not be found. This could mean that Neovim is not installed through bob.`
I have made sure that I don't have any other installation of neovim other than the one created by `bob`

**Comment by MordechaiHadad** [comment]
Do you remember what bob commands you used to install a neovim version?
and can you show me the content of `~/.local/share/bob`

**Comment by skyraider256526** [comment]
1. I have used `bob use stable` for installation
2. ~/.local/share/bob/
├── nvim-bin
│   └── nvim
├── stable
│   └── nvim-linux64
│       ├── bin
│       │   └── nvim
│       ├── lib
│       │   └── nvim
│       └── share
│           ├── applications
│           ├── icons
│           ├── locale
│           ├── man
│           └── nvim
└── used

**Comment by MordechaiHadad** [comment]
Can you show me the content of used, and what OS are u using?

**Comment by skyraider256526** [comment]
The `used` file contains `stable` and I am using `OS: Ubuntu 20.04.5 LTS x86_64`

**Comment by skyraider256526** [comment]
Okay, this was the result
>
┌───────────┬─────────────┐
│  Version  │  Status     │
├───────────┼─────────────┤
│  stable   │  Used       │
└───────────┴─────────────┘

**Comment by skyraider256526** [comment]
It's working without the ENV VARS

**Comment by skyraider256526** [comment]
okay I treid something different now, so I am running a script before trying to run nvim, if I don't run it it's working.
This is the script:
```
#!/usr/bin/sh

NVIM_BEGINNER=~/.config/nvim-beginner
export NVIM_BEGINNER

rm -rf $NVIM_BEGINNER

mkdir -p $NVIM_BEGINNER/share
mkdir -p $NVIM_BEGINNER/nvim

stow --restow --target=$NVIM_BEGINNER/nvim .

alias nvb='XDG_DATA_HOME=$NVIM_BEGINNER/share XDG_CONFIG_HOME=$NVIM_BEGINNER nvim' 

export nvb
```
I think the problem is because I am changing the `XDG_DATA_HOME` variable and the `used` file lies in `$XDG_DATA_HOME/bob`

**Comment by MordechaiHadad** [comment]
I am not very sure, bob just uses `~/.local/share/bob` without being dependent on `$XDG` env vars, so I am not sure why that would matter.
I can push a quick branch that will print the path, if u dont mind testing it.

**Comment by skyraider256526** [comment]
okay, I will try to test it

**Comment by MordechaiHadad** [comment]
once the build is complete download it and test it with the env vars
https://github.com/MordechaiHadad/bob/actions/runs/4302419812

**Comment by skyraider256526** [comment]
okay, could you write instructions I had installed `bob` using `cargo`

**Comment by MordechaiHadad** [comment]
copy this built bob to `~/.local/share/bob` as nvim(override existing nvim)
run `XDG_DATA_HOME=$NVIM_BEGINNER/share XDG_CONFIG_HOME=$NVIM_BEGINNER nvim `

**Comment by MordechaiHadad** [comment]
or if you dont know the location of bob simply run `bob eras` `bob use stable`

**Comment by MordechaiHadad** [comment]
Sorry, I meant copy to `~/.local/share/bob/nvim-bin`

**Comment by skyraider256526** [comment]
Okay I tried but it gave me following error:
```
nvim: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by nvim)
nvim: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by nvim)
nvim: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by nvim)
```
I just have one question, from the name nvim-bin it should have an nvim binary but you are telling me to replace the `nvim` binary in that folder with `bob` binary isn't this wrong ?

**Comment by MordechaiHadad** [comment]
Hmmm idk exactly about those neovim errors, maybe trying running the neovim binaries directly with those env vars and without and see if you get those errors.

Regarding your question, the exe in nvim-bin are actually an imposter, its actually a neovim binary but bob itself.

Recently bob moved to using proxy exe, so you are not actually running the neovim exe but the bob exe which since its named `nvim` it will direct the exe call to the nvim bin which its version is specified in the used file.

**Comment by MordechaiHadad** [comment]
@skyraider256526 I have created PR: #96 It's related to another issue but it might help with figuring out your issue

**Comment by MordechaiHadad** [comment]
@skyraider256526 have you managed to test the PR to see if it might fix your issue?

**Comment by MordechaiHadad** [comment]
@skyraider256526 anything?

**Comment by MordechaiHadad** [comment]
closing due to inactvity
